cmd/devenv/snapshot/snapshot.go

```diff
@@ -587,3 +640,175 @@ func (o *Options) CreateSnapshot(ctx context.Context) (string, error) {
+
+		err := uploadFromMinioToS3(ctx, source, sourceBucketName, itm.URI, o, s3c)
+		if err != nil {
+			return err
+		}
+
+		// Make this the latest version
+		destLockFile.TargetsV2[name].Snapshots[channel] = append(
+			[]*box.SnapshotLockListItem{itm}, destLockFile.TargetsV2[name].Snapshots[channel]...,
+		)
+	}
+
+	// upload latest yamls
+	byt, err := yaml.Marshal(destLockFile)
+	if err != nil {
+		return err
+	}
+
+	o.log.
+		WithField("bucket", o.b.DeveloperEnvironmentConfig.SnapshotConfig.Bucket).
+		WithField("key", lockFilePath).
+		Info("uploading latest yaml ...")
+
+	_, err = s3c.PutObject(ctx, &s3.PutObjectInput{
+		Bucket: &o.b.DeveloperEnvironmentConfig.SnapshotConfig.Bucket,
+		Key:    lockFilePath,
+		Body:   bytes.NewReader(byt),
+	})
+	if err != nil {
+		return err
+	}
+
+	// finally, delete snapshot infrastructure
+	err = snapshotstaging.CleanUp(ctx, o.log)
+	return err
+}
+
+func (o *Options) LoadLockFiles(ctx context.Context,
+	source *minio.Client,
+	sourceBucketName string,
+	lockFilePath *string, s3c *s3.Client) (sourceLockFile, destLockFile *box.SnapshotLock, err error) {
+	sourceLockFile = &box.SnapshotLock{}
+	sourceLockFileReader, err := source.GetObject(ctx, sourceBucketName, *lockFilePath, minio.GetObjectOptions{})
+	if err == nil {
+		defer sourceLockFileReader.Close()
+		if err := yaml.NewDecoder(sourceLockFileReader).Decode(&sourceLockFile); err != nil {
+			return nil, nil, errors.Wrap(err, "failed to parse remote snapshot lockfile")
+		}
+	} else {
+		return nil, nil, errors.Wrap(err, "failed to parse remote snapshot lockfile")
+	}
+
+	destLockFile = &box.SnapshotLock{}
+	resp, err := s3c.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: &o.b.DeveloperEnvironmentConfig.SnapshotConfig.Bucket,
+		Key:    lockFilePath,
+	})
+	if err == nil {
+		defer resp.Body.Close()
+		if err := yaml.NewDecoder(resp.Body).Decode(&destLockFile); err != nil {
+			return nil, nil, errors.Wrap(err, "failed to parse remote snapshot lockfile")
+		}
+	} else {
+		o.log.WithError(err).
+			Warn("Failed to fetch existing remote snapshot lockfile, will generate a new one")
+	}
+	return sourceLockFile, destLockFile, nil
+}
+
+func assureLockFilePath(destLockFile *box.SnapshotLock, name string, channel box.SnapshotLockChannel) {
+	if _, ok := destLockFile.TargetsV2[name]; !ok {
+		destLockFile.TargetsV2[name] = &box.SnapshotLockList{}
+	}
+
+	if destLockFile.TargetsV2[name].Snapshots == nil {
+		destLockFile.TargetsV2[name].Snapshots = make(map[box.SnapshotLockChannel][]*box.SnapshotLockListItem)
+	}
+
+	if _, ok := destLockFile.TargetsV2[name].Snapshots[channel]; !ok {
+		destLockFile.TargetsV2[name].Snapshots[channel] = make([]*box.SnapshotLockListItem, 0)
+	}
+}
+
+func uploadFromMinioToS3(ctx context.Context, source *minio.Client, sourceBucketName, key string, o *Options, s3c *s3.Client) error {
+	resp, err := source.GetObject(ctx, sourceBucketName, key, minio.GetObjectOptions{})
+
+	if err != nil {
```

Comment:
nit: no space between err assign and check

```suggestion
	resp, err := source.GetObject(ctx, sourceBucketName, key, minio.GetObjectOptions{})
	if err != nil {
```

Reply:
Same on L808